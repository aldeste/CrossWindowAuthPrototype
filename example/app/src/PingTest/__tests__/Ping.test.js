import React from "react";
import Ping from "../Ping";
import renderer from "react-test-renderer";

jest.mock("../../Document/Title").mock("../../Tags");

describe("PingTest / Ping", () => {
  global.fetch = () =>
    new Promise(resolve =>
      resolve({
        json: () => ({
          data: { person: { name: "Yoda" } },
          extensions: { timeTaken: "Time string" }
        })
      })
    );

  it("Matches previous setup", () => {
    const tree = renderer.create(<Ping />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Is a react component", () => {
    expect(typeof Ping).toBe("function");
    expect(<Ping />.$$typeof.toString()).toBe("Symbol(react.element)");
  });

  it("Fetches and updates state if button is clicked", async () => {
    const component = renderer.create(<Ping />);
    await component.getInstance().handleClick();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Prints out timeTaken", async () => {
    const component = renderer.create(<Ping />);
    await component.getInstance().handleClick();
    const tree = component.toJSON();
    expect(
      tree.children
        .filter(el => el.type === "p")
        .filter(el => el.children === "Time string")
    ).toBeDefined();
  });

  it("Prints out data", async () => {
    const component = renderer.create(<Ping />);
    await component.getInstance().handleClick();
    const tree = component.toJSON();
    expect(
      tree.children
        .filter(el => el.type === "p")
        .filter(el => el.children === "Yoda")
    ).toBeDefined();
  });

  it("Ussues callback on click", async () => {
    const callbackChecker = jest.fn();
    const component = renderer.create(<Ping callback={callbackChecker} />);
    await component.getInstance().handleClick();
    expect(callbackChecker).toHaveBeenCalled();
  });
});
