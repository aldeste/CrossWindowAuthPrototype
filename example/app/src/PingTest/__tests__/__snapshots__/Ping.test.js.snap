// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PingTest / Ping Fetches and updates state if button is clicked 1`] = `
<div
  className="View__View-jelnfZ bnYeJt"
>
  <h1
    className="Title__Title-lbDhID gxinMP"
  >
    Connection testing field
  </h1>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    This will ping a random user thowards the server. However. it won't sign in the user since the user isn't verefied
  </p>
  <button
    className="Button__Button-jkHDFw hKMOLu"
    onClick={[Function]}
  >
    Click to ping
  </button>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    Pinged with user 
    Yoda
  </p>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    The whole opperatiorn took 
    Time string
  </p>
</div>
`;

exports[`PingTest / Ping Matches previous setup 1`] = `
<div
  className="View__View-jelnfZ bnYeJt"
>
  <h1
    className="Title__Title-lbDhID gxinMP"
  >
    Connection testing field
  </h1>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    This will ping a random user thowards the server. However. it won't sign in the user since the user isn't verefied
  </p>
  <button
    className="Button__Button-jkHDFw hKMOLu"
    onClick={[Function]}
  >
    Click to ping
  </button>
</div>
`;
